tsdb/store.go

```diff
@@ -252,6 +261,7 @@ func (s *Store) CreateShard(database, retentionPolicy string, shardID uint64) er
 	}
 
 	s.shards[shardID] = shard
+	s.shardLocations[shardID] = &shardLocation{Database: database, RetentionPolicy: retentionPolicy, Shard: shard}
```

Comment:
Yeah that was my thinking, but I wanted to get feedback on whether it's OK to do away with `shards` first. Seems I forgot to even mention that in the PR description :smile:, so thanks for bring it up!

I'll refactor this morning.